reana_server/status.py

```diff
@@ -279,71 +278,57 @@ def __init__(self, from_=None, until=None, user=None):
-        )
-
-    def get_top_five_all(self, resource_type):
-        """Get the top five users according to quota usage."""
-        users = self.get_top_five_resource_usage_users(
-            Resource.type_ == resource_type, func.sum(UserResource.quota_used).desc()
-        )
-        return {
-            user.email: {
-                "id": str(user.id_),
+        return [
+            {
+                "email": user.user.email,
                 "used": ResourceUnit.human_readable_unit(
-                    user_resource.resource.unit, user_resource.quota_used
+                    user.resource.unit, user.quota_used
                 ),
                 "limit": ResourceUnit.human_readable_unit(
-                    user_resource.resource.unit, user_resource.quota_limit
+                    user.resource.unit, user.quota_limit
                 ),
+                "percentage": round(user.quota_used * 100.0 / user.quota_limit, 1),
             }
-            for user, user_resource in users
-        }
+            for user in users
+        ]
 
-    def get_top_five_limited(self, resource_type):
-        """Get the top five users with quota limit according to quota usage."""
-        users = self.get_top_five_resource_usage_users(
-            Resource.type_ == resource_type,
-            (
-                func.sum(UserResource.quota_used) / func.sum(UserResource.quota_limit)
-            ).desc(),
-            UserResource.quota_limit != 0,
+    def get_top_five(self, resource_type):
+        """Get the top five users according to quota usage."""
+        users = (
+            Session.query(UserResource)
+            .join(UserResource.resource)
+            .filter(Resource.type_ == resource_type)
+            .order_by(UserResource.quota_used.desc())
+            .limit(5)
         )
-        return {
-            user.email: {
-                "id": str(user.id_),
+        return [
+            {
+                "email": user.user.email,
                 "used": ResourceUnit.human_readable_unit(
-                    user_resource.resource.unit, user_resource.quota_used
+                    user.resource.unit, user.quota_used
                 ),
                 "limit": ResourceUnit.human_readable_unit(
-                    user_resource.resource.unit, user_resource.quota_limit
+                    user.resource.unit, user.quota_limit
                 ),
             }
-            for user, user_resource in users
-        }
+            for user in users
+        ]
 
     def get_status(self):
         """Get status summary for REANA system."""
         return {
-            "top_five_all_disk": self.get_top_five_all(ResourceType.disk),
-            "top_five_all_cpu": self.get_top_five_all(ResourceType.cpu),
+            "top_five_disk": self.get_top_five(ResourceType.disk),
+            "top_five_cpu": self.get_top_five(ResourceType.cpu),
             "top_five_limited_disk": self.get_top_five_limited(ResourceType.disk),
```

Comment:
Works well :+1: Tested in debugging mode, not in production mode.

One nomenclature comment. `top_five_disk` and `top_five_cpu` lists all the users irrespective of quotas.  This is good. `top_five_limited_disk` and `top_five_limited_cpu` is virtually the same, only sorted by percentage.  This is because users without limit are dividing by infinity, so stay close to zero, as it were.  Perhaps you can rename the headings to top_five_disk_percentage` and `top_five_cpu_percentage`, which might be clearer?

On a similar tune, we could display the percentage information also in the first table; only it won't be sorted by that column.

Finally, a cosmetic comment. Since we display units inside table, we could treat the percentage the same, and say:

```
  EMAIL             | USED    | LIMIT   | PERCENTAGE
  test3@example.org | 199 MiB | 200 MiB | 99.5%
  test4@example.org |  90 KiB | 100 KiB | 90.0%
  test1@example.org |  42 MiB |  50 MiB | 84.0%
  test2@example.org |  20 MiB | 100 MiB | 20.0%
```

Also, you could right-align columns for even nicer cosmetics.

Reply:
Latest quota usage output:
```
QUOTA-USAGE
top_five_disk:
  EMAIL                      |    USED |   LIMIT | PERCENTAGE
  test3@example.org          | 100 MiB | 200 MiB |      50.0%
  test1@example.org          |  42 MiB |  50 MiB |      84.0%
  test2@example.org          |  20 MiB | 100 MiB |      20.0%
  test4@example.org          |  90 KiB | 100 KiB |      90.0%
  audrius.mecionis@gmail.com | 0 Bytes | 0 Bytes |           

top_five_cpu:
  EMAIL                      |   USED |  LIMIT | PERCENTAGE
  test4@example.org          | 1m 35s | 1m 40s |      95.0%
  test2@example.org          | 1m 30s |    15m |      10.0%
  test1@example.org          |    25s |    50s |      50.0%
  test3@example.org          |    20s |     5h |       0.1%
  audrius.mecionis@gmail.com |     0s |     0s |           

top_five_disk_percentage:
  EMAIL             |    USED |   LIMIT | PERCENTAGE
  test4@example.org |  90 KiB | 100 KiB |      90.0%
  test1@example.org |  42 MiB |  50 MiB |      84.0%
  test3@example.org | 100 MiB | 200 MiB |      50.0%
  test2@example.org |  20 MiB | 100 MiB |      20.0%

top_five_cpu_percentage:
  EMAIL             |   USED |  LIMIT | PERCENTAGE
  test4@example.org | 1m 35s | 1m 40s |      95.0%
  test1@example.org |    25s |    50s |      50.0%
  test2@example.org | 1m 30s |    15m |      10.0%
  test3@example.org |    20s |     5h |       0.1%
```

Reply:
When users don't have quotas, the "top_five_disk" table looks OK, but "top_five_cpu" table is not well aligned for me. Can you please have a look?

```
QUOTA-USAGE
top_five_disk:
  EMAIL               |       USED |   LIMIT | PERCENTAGE
  tibor.simko@cern.ch |   1.01 MiB | 0 Bytes |           
  test1@example.org   | 174.85 KiB | 0 Bytes |           
  test2@example.org   |    0 Bytes | 0 Bytes |           
  test3@example.org   |    0 Bytes | 0 Bytes |           
  test4@example.org   |    0 Bytes | 0 Bytes |           

top_five_cpu:
  EMAIL               |   USED | LIMIT | PERCENTAGE
  tibor.simko@cern.ch | 2m 26s | 0s |           
  test1@example.org   |    27s | 0s |           
  test2@example.org   |     0s | 0s |           
  test3@example.org   |     0s | 0s |           
  test4@example.org   |     0s | 0s |  
```

Reply:
... but actually we can merge anyway, this is just cosmetics that we may need to amend in real-life conditions as the values will be accumulating, for example:

```
top_five_cpu:
  EMAIL               |   USED |   LIMIT | PERCENTAGE
  test4@example.org   | 22800h | 240000h |       9.5%
  test3@example.org   |   480h | 432000h |       0.1%
```